Implementing backup to S3 Java example function

Shylajhaa opened the pull request:
Implementing backup to S3 Java example function

1. Created a function in project
<img width="718" alt="Screenshot 2021-10-28 at 10 28 33 PM" src="https://user-images.githubusercontent.com/22257998/139302735-3a5dc88f-d6eb-44e0-89f0-22727e2ed8e4.png">

2. Execute function

3. On successful execution message is printed to console
<img width="926" alt="Screenshot 2021-10-28 at 10 28 49 PM" src="https://user-images.githubusercontent.com/22257998/139302810-4f0db36d-4524-4017-bd74-cbc2a94650bc.png">

4. File is uploaded to S3
<img width="999" alt="Screenshot 2021-10-30 at 3 25 49 PM" src="https://user-images.githubusercontent.com/22257998/139528420-b2afd46b-7b65-455d-8a41-707281631d26.png">

Shylajhaa commented:
@PineappleIOnic Made the necessary changes. Please check now

Meldiron commented:
Connecting this PR to issue https://github.com/appwrite/appwrite/issues/1890

Meldiron commented on java/backup-to-s3/src/main/java/BackupToS3.java:
```diff
@@ -0,0 +1,73 @@
+            collectionsCsv.append(headers);
+            collectionsCsv.append('\n');
+
+            HttpResponse<JsonNode> response =
+                    Unirest.get(String.format("%s/database/collections", System.getenv("APPWRITE_ENDPOINT")))
+                            .header("X-Appwrite-Project", System.getenv("APPWRITE_FUNCTION_PROJECT_ID"))
+                            .header("X-Appwrite-key", System.getenv("APPWRITE_API_KEY"))
+                            .asJson();
```
Please let's use Appwrite library so we don't have to struggle with building the HTTP request. Also, please add pagination as this would only fetch first 25 collections

Shylajhaa replied:
@Meldiron I have worked on pagination. However while using AppwriteClient I'm facing issues discussed in this thread - https://discord.com/channels/564160730845151244/564161373148414012/914957241436749844

Can I just go ahead with using Unirest?

Shylajhaa replied:
cc: @christyjacob4

Meldiron commented on java/backup-to-s3/src/main/java/BackupToS3.java:
```diff
@@ -0,0 +1,73 @@
+            for (int i=0; i < collections.length(); i++) {
+                JSONObject jsonCollection = collections.getJSONObject(i);
+
+                collectionsCsv.append(jsonCollection.getString("$id"));
+                collectionsCsv.append(",");
+                collectionsCsv.append(jsonCollection.getJSONObject("$permissions").toString());
+                collectionsCsv.append(",");
+                collectionsCsv.append(jsonCollection.getString("name"));
+                collectionsCsv.append(",");
+                collectionsCsv.append((char) jsonCollection.getLong("dateCreated"));
+                collectionsCsv.append(",");
+                collectionsCsv.append((char) jsonCollection.getLong("dateUpdated"));
+                collectionsCsv.append(",");
+                collectionsCsv.append(jsonCollection.getJSONArray("rules").toString());
+                collectionsCsv.append("\n");
```
We are storing information (metadata) about the collection, but we are not storing data themselves. For that, we need to use "listDocuments". Also, let's make sure to implement pagination, otherwise, it would only backup 25 documents.

Meldiron reviewed (changes requested):
Requested few changes.

christyjacob4 reviewed (changes requested):
I have similar comments to that in your Kotlin PR. Please fix those 👍

Shylajhaa commented:
> I have similar comments to that in your Kotlin PR. Please fix those 👍

@christyjacob4 Addressed PR comments except moving to Appwrite Client, Have hit APIs using Unirest

christyjacob4 commented:
THANK YOU! All changes merged 🥳

Please reach out to me on discord to claim your Hacktoberfest swag.

As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord so that we can add you as a contributor to our organization.

Once added, you can change your membership visibility settings, so your new Appwrite membership badge will show up on your personal GitHub profile.
